contracts/basic-deal-client/DealClient.sol

```diff
@@ -63,64 +98,72 @@ contract DealClient {
         owner = msg.sender;
     }
 
-    // FOR DEMO PURPOSES
-    function simpleDealProposal(bytes memory pieceCid, uint64 pieceSize) public {
-        ContractDealProposal memory deal;
-        deal.piece_cid = pieceCid;
-        deal.piece_size = pieceSize;
-        deal.storage_price_per_epoch = uintToBigInt(0);
-
-        bytes memory serDeal = serializeContractDealProposal(deal);
-        makeDealProposal(serDeal);
-
-    }
-
-    function makeDealProposal(bytes memory deal) public {
+    function makeDealProposal(DealRequest calldata deal) public {
         // TODO: evaluate permissioning here
         require(msg.sender == owner);
 
+        uint256 index = deals.length;
+        deals.push(deal);
+
         // creates a unique ID for the deal proposal -- there are many ways to do this
-        bytes32 _id = keccak256(abi.encodePacked(block.timestamp, msg.sender));
-        dealProposals[_id] = deal;
+        bytes32 _id = keccak256(abi.encodePacked(block.timestamp, msg.sender, index));
+        dealProposals[_id] = index;
 
-        ContractDealProposal memory proposal = deserializeContractDealProposal(deal);
-        pieceToProposal[proposal.piece_cid] = ProposalIdSet(_id, true);
+        pieceToProposal[deal.piece_cid.data] = ProposalIdSet(_id, true);
 
         // writes the proposal metadata to the event log
-        emit DealProposalCreate(_id, proposal.piece_size, proposal.verified_deal, 0); //bigIntToUint(proposal.storage_price_per_epoch));
+        emit DealProposalCreate(_id, deal.piece_size, deal.verified_deal, 0); //bigIntToUint(deal.storage_price_per_epoch));
```

Comment:
change the last arg here to
`deal.storage_price_per_epoch`

Reply:
Done!